src/Core/System/DependencyInjection/CompilerPass/SalesChannelEntityCompilerPass.php

```diff
@@ -116,6 +116,8 @@ private function collectDefinitions(ContainerBuilder $container): void
             if (isset($entityNames['fallBack'])) {
                 $repositoryNameMap[$entityNames['fallBack']] = $repositoryId;
             }
+
+            $container->registerAliasForArgument(SalesChannelRepository::class, $entityNames['entityName'] . '.repository', $repositoryId);
```

Comment:
To be consistent i would also add a alias for `SalesChannelRepositoryInterface` for each repo

Reply:
i test it. it is possible to use this or the other interface,
but if a plugin is not compatible, you can't use both.

It should be fast removed from plugins.

Reply:
thats fine that we add only for the new one. So we force the people to use the newer classes